src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,1441 @@
+        visualizer.property11 = Property.None
+        visualizer.property12 = Property.None
+    }
+
+    if (visualizer.preset == Preset.Primes) {
+        visualizer.backgroundColor = BLACK
+        visualizer.property1 = Property.Prime
+        visualizer.property1Visualization = PropertyVisualization.Color
+        visualizer.property1MainColor = RED
+    } else if (visualizer.preset == Preset.Abundant_Numbers) {
+        visualizer.backgroundColor = WHITE
+        visualizer.property1 = Property.Abundant
+        visualizer.property1Visualization = PropertyVisualization.Color
+        visualizer.property1MainColor = BLACK
+    } else if (visualizer.preset == Preset.Abundant_Numbers_And_Primes) {
+        visualizer.backgroundColor = WHITE
+        visualizer.property1 = Property.Abundant
+        visualizer.property1Visualization = PropertyVisualization.Color
+        visualizer.property1MainColor = BLACK
+        visualizer.property2 = Property.Prime
+        visualizer.property2Visualization = PropertyVisualization.Color
+        visualizer.property2MainColor = RED
+    } else if (visualizer.preset == Preset.Polygonal_Numbers) {
+        visualizer.backgroundColor = BLACK
+        visualizer.property1 = Property.Triangular_Number
+        visualizer.property1Visualization = PropertyVisualization.Color
+        visualizer.property1MainColor = RED
+        visualizer.property2 = Property.Square_Number
+        visualizer.property2Visualization = PropertyVisualization.Color
+        visualizer.property2MainColor = ORANGE
+        visualizer.property3 = Property.Pentagonal_Number
+        visualizer.property3Visualization = PropertyVisualization.Color
+        visualizer.property3MainColor = YELLOW
+        visualizer.property4 = Property.Hexagonal_Number
+        visualizer.property4Visualization = PropertyVisualization.Color
+        visualizer.property4MainColor = GREEN
+        visualizer.property5 = Property.Heptagonal_Number
+        visualizer.property5Visualization = PropertyVisualization.Color
+        visualizer.property5MainColor = BLUE
+        visualizer.property6 = Property.Octagonal_Number
+        visualizer.property6Visualization = PropertyVisualization.Color
+        visualizer.property6MainColor = PURPLE
+    }
+}
+
+function setOverridingSettings(visualizer: VisualizerGrid) {
+    if (visualizer.resetAndAugmentByOne) {
+        visualizer.pathType = PathType.Rows
+    }
+
+    if (
+        (visualizer.visualInfo == VisualInfo.Show_Numbers
+            || visualizer.visualInfo == VisualInfo.Show_Path)
+        && visualizer.amountOfNumbers > 400
+    ) {
+        visualizer.amountOfNumbers = 400
+    }
+
+    if (
+        visualizer.formula == Formula.TWO_TO_THE_N
+        && visualizer.amountOfNumbers > 1000
+    ) {
+        visualizer.amountOfNumbers = 1000
+    }
+}
+
+function getNumberFromFormula(
+    visualizer: VisualizerGrid,
+    currentSequenceIndex: number,
+    augmentForSequence: number
```

Comment:
As far as I can tell, the purpose of getNumberFromFormula is to transform the input sequence that the visualizer is being run on by a further function. In other words, if the sequence specified in the UI was A000045, the Virahanka-Fibonacci numbers, but you really wanted the squares of the V-F numbers, you would visualize the OEIS sequence A000045 and set the "formula" parameter to N_SQUARED, correct? There is a planned feature that will allow formulas with OEIS references mixed in that will make it unnecessary to allow for such things, because you will be able to write for example `A000045^2`.

The moral of that story, though, is that Visualizers should be set up just to use the values of a sequence as they come (you can add the row augmentation after you get the sequence value, of course).

In the meantime, given that the mixed OEIS/formula sequences are not available at the moment, if you really want this "transformation" ability right now, it's something that's entirely happening on the Sequence side, so the facility should be moved to a new Sequence class rather than mixing Sequence computation into the internals of a Visualizer. Not only will that simplify the code here, but it will allow _anyone_'s visualizer to use a transformed OEIS sequence immediately, not just yours.

If you want to do that, you should modify `src/sequences/OEISSequenceTemplate.ts` to replace the "mod" parameter with a text box to enter an arbitrary (mathjs) formula to transform the OEIS sequence by, and then use mathjs to parse that formula into a function that you will evaluate on the OEIS sequence values -- see `src/sequences/SequenceFormula.ts` for how to do this. You should accept any formula that has a single free variable. That way, to get the squares of the V-F numbers, you would enter A000045 for the sequence ID and say `x^2` for the transformation. And to recover the previous facility of modding out by 5 (say), you would just use `mod(x, 5)` for the transformation. And all of your examples below are easy to write as mathjs formulas, and then you won't anywhere need a laundry list of different possible transformations that is always doomed not to have the one that the person using the visualizer wants in that instance (which is I presume how you ended up with this particular hardcoded collection of formulas to select from).

Reply:
Sorry if I was too long-winded/unclear above. Visualizers should not be transforming Sequence values. There is a separation of concerns between generating sequence values and visualizing them between the Sequence and Visualizer class hierarchy. So if you need any/all of the transforms you had, they should be moved into the OEISSequenceTemplate class (and implemented with mathjs formulas, since we already have that). Thanks.

Reply:
I did not end up making the advised change to SequenceForumla.ts. I feel that squaring the numbers is one of the "core ideas" of the visualizer but that all other formulas are unnecessary. I also feel that it would be quite difficult for me to figure out how to make the advised change. It is okay for me just to square sequence values just for one case?

Reply:
Well, this needs to be moved to the Sequence side for this PR to go in, so I will do that once this branch is up to date with main.